File: programmers/기초/문자열 여러 번 뒤집기.js
// 문자열 여러 번 뒤집기
// queries의 원소는 [s, e] 형태로, my_string의 인덱스 s부터 인덱스 e까지를 뒤집으라는 의미
function solution(my_string, queries) {
  let arr = my_string.split("");
  let reverse = "";
  for (let x of queries) {
    let num = x[1] - x[0];
    reverse = arr.slice(x[0], x[1] + 1).reverse();

    arr.splice(x[0], num + 1);
    arr.splice(x[0], 0, ...reverse);
  }
  return arr.join("");
}

File: programmers/python/문자열 여러 번 뒤집기.py
# 문자열 여러 번 뒤집기
def solution(str, q):
    str = list(str)
    for s, e in q:
        str[s:e+1] = reversed(str[s:e+1])
    return ''.join(str)

def solution(my_string, queries):
    for s, e in queries:
        my_string = my_string[:s]+my_string[s:e+1][::-1]+my_string[e+1:]

    return my_string



